include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -185,26 +185,46 @@ using __kernel_name_generator =
     _BaseName<_CustomName>;
 #endif
 
-template <typename _DerivedKernelName>
+template <typename... _KernelNames>
 class __kernel_compiler
 {
+    using _KernelArray = std::array<sycl::kernel, sizeof...(_KernelNames)>;
+    _KernelArray __m_kernels;
+
+#if _ONEDPL_KERNEL_BUNDLE_PRESENT
+    sycl::kernel_bundle<sycl::bundle_state::executable> __m_kernel_bundle;
+#endif
+
   public:
+#if _ONEDPL_KERNEL_BUNDLE_PRESENT
     template <typename _Exec>
-    static sycl::kernel
```

Comment:
I would assume it's better to create return either `sycl::kernel_bundle` or sycl::kernel depending on modern or deprecated API rather than trying to kernel bundle in the layout conditionally. Seems like we also need kernels array. We might think about returning a special struct or `tuple`

Reply:
In both cases we need sycl::kernel. It is used for wg size calculation, see __kernel_work_group_size, __kernel_sub_group_size.

Reply:
The second variant was done -  a method `compile` returns "kernels array or just a kernel".